add amazon reviews

joeddav opened the pull request:
Adds the Amazon US Reviews dataset as requested in #353. Converted from [TensorFlow Datasets](https://www.tensorflow.org/datasets/catalog/amazon_us_reviews). cc @clmnt @sshleifer

patrickvonplaten commented on datasets/amazon_us_reviews/amazon_us_reviews.py:
```diff
@@ -0,0 +1,199 @@
+    review_date       - The date the review was written.
+"""
+
+_DATA_OPTIONS_V1_00 = [
```
Do we need `_DATA_OPTIONS_V1_00`, ... really? Since it seems that all of _00, _01, _02, is concatenated anyways and never used on its own, I'd think that it's better to just have one `_PRODUCTS` list

patrickvonplaten commented on datasets/amazon_us_reviews/amazon_us_reviews.py:
```diff
@@ -0,0 +1,199 @@
+class AmazonUSReviewsConfig(datasets.BuilderConfig):
+    """BuilderConfig for AmazonUSReviews."""
+
+    def __init__(self, *, data=None, **kwargs):
```
why do we have the `*` in the init?

thomwolf replied:
Usually this is to prevent the use of positional args (we used that for the metrics in the lib to prevent user-mistakes) but I'll let @joeddav confirm

joeddav replied:
This is just an artifact I missed from the original TFDS code. Maybe they used it for that reason but I don't think it's needed.

patrickvonplaten commented on datasets/amazon_us_reviews/amazon_us_reviews.py:
```diff
@@ -0,0 +1,199 @@
+        path = dl_manager.download_and_extract(url)
+
+        # There is no predefined train/val/test split for this dataset.
+        return [
```
I'm fine with this! I did the splitting myself for other datasets, just (90%, 5%, 5%), but not at all sure what the better approach is here. Think this way here is actually better because if there is no official split we should not do it and the user can easily make use of `select` or `filter` to do the split her/himself.

joeddav replied:
Yeah my feeling is if there's no official split that it's best to let the user create their own, but I guess that's a general question that we don't have a consensus around.

patrickvonplaten reviewed (approved):
PR looks good to me! All the slow tests pass. 

I added some suggestions, but I don't think there are minor things. 
Still curious though why there is a `*` in the `__init__` of the config.

joeddav commented:
@patrickvonplaten Yeah this is adapted from tfds so a lot is just how they wrote the code. Addressed your comments and also simplified the weird `AmazonUSReviewsConfig` definition. Will merge once tests pass.

lhoestq commented:
Thanks for checking this one :) 
Looks good to me 

Just one question : is there a particular reason to use `names=["Y", "N"]` in this order ? Usually the positive label is at index 1 and the negative one at index 0 for binary classification

joeddav commented:
> is there a particular reason to use `names=["Y", "N"]` in this order ? Usually the positive label is at index 1 and the negative one at index 0 for binary classification

Hmm that's a good point. I'll submit a quick fix.
